### PR TITLE
Add learning outcomes and knowledge checks to JavaScript/More Testing lesson

### LINKED ISSUE #22804: Add missing Knowledge Checks, Learning Outcomes & answer-links to JavaScript/More Testing
Related to #22472

This task is to go through the lesson referenced in the title of this issue and (as applicable):

- Add missing Learning Outcomes
- Add missing Knowledge Checks
- Add answer links in the Knowledge Checks

If the lesson already has Learning Outcomes or Knowledge Checks they do not need to be added to. Many lessons only need answer links added.

Knowledge check links should consist of an `<a>` tag which adds the `knowledge-check-link` class and links to the part of the lesson (or the external resource that is linked in the lesson) where the answer to the Knowledge Check can be found. Knowledge Check links should **not** point to external resources that weren't linked to as part of the lesson content or assignment. You can link to a specific line in the lesson like so:

```html
Lorem ipsum dolor sit amet, <span id="js-function-knowledge-check">consectetur adipiscing elit</span>, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to the page where the span with this id lives -->
<a class="knowledge-check-link" href="#js-function-knowledge-check">How do you return a value from a function?</a>
```
Or you can link to just a section of the lesson by just linking to the section heading and following [these](https://github.com/TheOdinProject/curriculum/wiki/Contributing-to-Wiki#anchor-linking) rules for determining how to format the link. Example:

```html
### Lorem ipsum dolor sit amet

sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to a heading on the page-->
<a class="knowledge-check-link" href="#lorem-ipsum-dolor-sit-amet">How do you return a value from a function?</a>
```

### Acceptance Criteria

- [ ] Lesson referenced in the title has Learning Outcomes, Knowledge Checks, and Knowledge Check links
- [ ] Knowledge Check links all link to the proper place in the lesson, or the proper external resource that was linked to earlier in the lesson
- [ ] Knowledge Check styling should **not** change

Please do not start work on this unless you've been assigned to it. Comment on this issue and ping me @wise-king-sullyman if you would like to be assigned to it.

Additionally, we would like for contributors to only take on one of these issues at a time. Please only request assignment for (and begin work on) an issue after the Learning Outcome/Knowledge Check/Knowledge Check link issue you're currently assigned to has been approved and merged.

Please check out [this new post on our wiki](https://github.com/TheOdinProject/curriculum/wiki/Learning-Outcomes-vs-Knowledge-Checks) about Learning Outcomes and Knowledge Checks if you need to write new Learning Outcomes or Knowledge Checks for this issue.

### PR DESCRIPTION
**Pull request by rusty-reebs**
- [X] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [X] The title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 - [X] If the PR is related to an open issue, use a [relevant keyword](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) and reference it with the `#` sign and the issue number.
 - [X] You have previewed your markdown (if any) in the [Odin Markdown Preview Tool](https://www.theodinproject.com/lessons/preview)
 - [X] If changes are requested, please make the changes in a timely manner. After you submit the changes, request another review from the maintainer (top right).

#### 1. Describe the changes made and include why they are necessary or important:

Write learning outcomes and knowledge checks for JavaScript/More Testing lesson.

#### 2. Related Issue

Closes #22804

### CODE REVIEW

File: javascript/testing/testing-2.md [hunk 1]
```diff
@@ -63,3 +70,12 @@ There are two solutions to the 'tightly coupled code' problem.  The first, and b
 4. Jest includes some _really_ handy mocking functions.  Read about them in the [official docs](https://facebook.github.io/jest/docs/en/mock-functions.html#content)
 5. And finally, if you wish, you can add Jest to your webpack setup.  Read about that process [here](https://facebook.github.io/jest/docs/en/webpack.html#content).
 </div>
+
+### Knowledge Check 
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class='knowledge-check-link' href='#tightly-coupled-code'>What is tightly coupled code?</a>
+- <a class='knowledge-check-link' href='https://medium.com/@jamesjefferyuk/javascript-what-are-pure-functions-4d4d5392d49c'>What are the two requirements for a function to be pure?</a>
+- <a class='knowledge-check-link' href='https://medium.com/@jamesjefferyuk/javascript-what-are-pure-functions-4d4d5392d49c'>What is an observable side effect in regards to a function?</a>
```
Comment (fortypercenttitanium) [inline javascript/testing/testing-2.md:79]: Can you rephrase this question? If I'm a learner I'd be confused if "observable side effect" was a technical term. I'd also be a bit confused on what you mean by "in regards to a function".

File: javascript/testing/testing-2.md [hunk 2]
```diff
@@ -63,3 +70,12 @@ There are two solutions to the 'tightly coupled code' problem.  The first, and b
...
+- <a class='knowledge-check-link' href='#two-solutions'>What are two solutions to the tightly coupled code problem?</a>
+- <a class='knowledge-check-link' href='#two-solutions'>What is mocking?</a>
```
Comment (fortypercenttitanium) [inline javascript/testing/testing-2.md:81]: Can you add one about when you'd want to use mocking?

**Review by fortypercenttitanium — CHANGES_REQUESTED** [review]
Great work on this, thanks. Check out my comments and let me know what you think.

File: javascript/testing/testing-2.md (hunk 1 above)
Comment (rusty-reebs) [reply javascript/testing/testing-2.md]: Yes, no problem.

File: javascript/testing/testing-2.md (hunk 2 above)
Comment (rusty-reebs) [reply javascript/testing/testing-2.md]: Done. :)

File: javascript/testing/testing-2.md (hunk 1 above)
Comment (fortypercenttitanium) [reply javascript/testing/testing-2.md]: I'm thinking like something along the lines of "What is a "side effect", and why is it important to identify them when testing a function?"

Comment (rusty-reebs) [reply javascript/testing/testing-2.md]: Ok, got it. 👍

**Review by fortypercenttitanium — APPROVED** [review]
Nice job! Thank you for your patience.
